3_visualize/src/drought_learner_viz_components.R

```diff
@@ -885,5 +891,153 @@ frame_l <- function(p2_streamflow_learner_viz_df, p2_droughts_learner_viz_df,
+  p2_droughts_site_wrapYr_endOfYear <- p2_droughts_site_wrapYr %>%
+    mutate(end = as.Date(sprintf("%s-12-31", year(start))))
+  p2_droughts_site_wrapYr <- bind_rows(p2_droughts_site_wrapYr_beginningOfYear,
+                                       p2_droughts_site_wrapYr_endOfYear) 
+  #   3. Fix metadata
+  p2_droughts_jd_wrapYr <- p2_droughts_jd_wrapYr %>%
+    mutate(start_year = year(start),
+           start_noYr = format(as.Date(start), "%m-%d"),
+           start_fakeYr = as.Date(sprintf("1963-%s", start_noYr)),
+           end_year = year(end),
+           end_noYr = format(as.Date(end), "%m-%d"),
+           end_fakeYr = as.Date(sprintf("1963-%s", end_noYr)))
+  p2_droughts_site_wrapYr <- p2_droughts_site_wrapYr %>%
+    mutate(start_year = year(start),
+           start_noYr = format(as.Date(start), "%m-%d"),
+           start_fakeYr = as.Date(sprintf("1963-%s", start_noYr)),
+           end_year = year(end),
+           end_noYr = format(as.Date(end), "%m-%d"),
+           end_fakeYr = as.Date(sprintf("1963-%s", end_noYr)))
+  #   4. Merge back in with other records
+  p2_droughts_jd_noWrapYr <- p2_droughts_jd %>%
+    filter(start_year == end_year)
+  p2_droughts_jd <- bind_rows(p2_droughts_jd_wrapYr,
+                              p2_droughts_jd_noWrapYr)
+  p2_droughts_site_noWrapYr <- p2_droughts_site %>%
+    filter(start_year == end_year)
+  p2_droughts_site <- bind_rows(p2_droughts_site_wrapYr,
+                                p2_droughts_site_noWrapYr)
+  
+  
+  blank_plot_year <- ggplot(data = p2_streamflow_learner_viz_df, aes(y = value, x = dt))+
+    ylab("Year\n(sort of)")+
+    xlab(NULL)+
+    scale_x_date(labels = date_format("%b"), 
+                 date_breaks  ="1 month",
+                 limits = c(as.Date("01/01/1963",'%d/%m/%Y'), as.Date("31/12/1963",'%d/%m/%Y')))+
+    theme_tufte(base_family = "sans", base_size = 16)+
+    theme(axis.line = element_line(color = 'black'),
+          axis.text = element_text(size = 6),
+          axis.title = element_text(size = 8),
+          panel.background = element_blank())
+  
+  main <- blank_plot_year +
+    # Fixed threshold drought durations
+    annotate("rect", # fixed threshold
+             xmin = (p2_droughts_site$start_fakeYr),
+             xmax = (p2_droughts_site$end_fakeYr),
+             ymin = p2_droughts_site$start_year-1, ymax = p2_droughts_site$start_year,
+             fill = "#da7d81", alpha = 0.9) +
+    # Variable threshold drought durations
+    annotate("rect", # variable threshold
+             xmin = (p2_droughts_jd$start_fakeYr),
+             xmax = (p2_droughts_jd$end_fakeYr),
+             ymin = p2_droughts_jd$start_year-70, ymax = p2_droughts_jd$start_year-69,
+             fill = "#e6af84", alpha = 0.9)+
+    theme(axis.title.y = element_text(color="transparent"),
+          axis.text.y = element_text(color = "transparent"),
+          axis.ticks.y = element_line(color = "transparent"))
+  
+  
+  # Bottom bars
```

Comment:
RC: Maybe rename these now that they are no longer bottom bars

Reply:
These bottom bars are still underneath this image as bottom bars, although not sure what value they add to these later viz frames

Reply:
oooh sorry I mistook this section. Looks like they aren't actually added to the plot, so this section could maybe be deleted? At least I don't see them specified in lines 1017-1039 and don't see them in the plot on the website:
![image](https://user-images.githubusercontent.com/54007288/185680433-5e9c9e4b-e87f-4348-9108-0ac04c35f307.png)

Reply:
They were supposed to still be on there for now, so I fixed it so they show correctly. Thanks for noticing!